libs/traits/src/lib.rs

```diff
@@ -135,18 +135,35 @@ pub trait PoolInspect<AccountId, CurrencyId> {
 	/// check if the pool exists
 	fn pool_exists(pool_id: Self::PoolId) -> bool;
 	fn tranche_exists(pool_id: Self::PoolId, tranche_id: Self::TrancheId) -> bool;
-	fn get_tranche_token_price(
-		pool_id: Self::PoolId,
-		tranche_id: Self::TrancheId,
-	) -> Option<PriceValue<CurrencyId, Self::Rate, Self::Moment>>;
 
 	/// Get the account used for the given `pool_id`.
 	fn account_for(pool_id: Self::PoolId) -> AccountId;
 
-	// Get the currency used for the given `pool_id`.
+	/// Get the currency used for the given `pool_id`.
 	fn currency_for(pool_id: Self::PoolId) -> Option<CurrencyId>;
 }
 
+/// Get the latest price for a given tranche token
+pub trait TrancheTokenPrice<AccountId, CurrencyId> {
+	type PoolId: Parameter
+		+ Member
+		+ Debug
+		+ Copy
+		+ Default
+		+ TypeInfo
+		+ Encode
+		+ Decode
+		+ MaxEncodedLen;
+	type TrancheId: Parameter + Member + Debug + Copy + Default + TypeInfo + MaxEncodedLen;
```

Comment:
Open a discussion because I always have the same dilemma. Should we add these bounds here? They seem to be required only for a specific implementation (for example, an implementation where `PoolId` is encoded in a database). On the other hand, I know that adding these bounds here is easy to handle on the pallet side because you do not need to add some `where` bounds.

Reply:
Thanks for raising this point. I had the same thought to be honest. For me, the least amount of "things" the best since that means less clutter, less inter dependencies, and less uneccessary code to maintain. I'll have a further look at this and how we can reduce its footprint

Reply:
> Thanks for raising this point. I had the same thought to be honest.

Same here! I should have raised that question earlier. I am on the same page as Nuno.

Reply:
Done 👍